### PR TITLE
Feature/implement tracking of tags in 3 d model

### PR DESCRIPTION
**Pull request by Noggling**
# Description

Overall 3d Refactor, to prevent function firing at strange times when selecting. Adding Electro tags with status. Status has to be verified by Atle

Completes: [AB#214110](https://dev.azure.com/Equinor/fa63ceea-8883-41e2-8823-a3b54e4be178/_workitems/edit/214110) 

## Checklist:

-   [x] I have performed a self-review of my own code.
-   [x] I have linked my DevOps task using the AB# tag.
-   [x] My code is easy to read, and comments are added where needed.
-   [ ] My code is covered by tests.

### CODE REVIEW

File: packages/echo3dViewer/src/services/echo3dMultiSelectionHandler.ts [hunk 1]
```diff
@@ -183,7 +189,33 @@ export class Echo3dMultiSelectionHandler extends Echo3dBaseSelectionHandler {
...
+     * @param {HierarchyNodeModel[]} [hierarchyNodeModel] Nodes for abb and tag
+     * @memberof Echo3dSelectionHandler
+     */
+    private addSelectedNodesToViewerSelection(hierarchyNodeModel?: HierarchyNodeModel[]): void {
+        const selection: {position: THREE.Vector3, tagNo: string}[] = []
+            
+        if (hierarchyNodeModel) {
+                hierarchyNodeModel.forEach((nodeResult) => {
+        
+                    if (nodeResult.aabb && nodeResult.tag ) {
```
Comment (Gustav-Eikaas) [inline packages/echo3dViewer/src/services/echo3dMultiSelectionHandler.ts:211]: Could have just filter then map, but its correct

File: src/packages/ModelViewer/hooks/useOverlay.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        }[]
+    >([]);
+    const handleOnCameraChange = useCallback(() => {
+        if (echo3DClient) {
+            const camera = echo3DClient.viewer.getCamera();
+            const renderer = echo3DClient.viewer.renderer;
+
+            const s: {
+                tagNo: string;
+                key: string;
```
Comment (Gustav-Eikaas) [inline src/packages/ModelViewer/hooks/useOverlay.ts:23]: INTERFACE !

File: src/packages/ModelViewer/hooks/useOverlay.ts [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+                domPosition: { left: string; top: string };
+                position: THREE.Vector2 | undefined;
+            }[] = [];
+
+            viewerSelection.forEach((vs, i) => {
+                if (camera && renderer) {
+                    const position = getDomPositionFor3DPosition(camera, renderer, vs.position);
+                    if (position) {
+                        position.x += renderer.domElement.offsetLeft;
+                        position.y += renderer.domElement.offsetTop;
```
Comment (Gustav-Eikaas) [inline src/packages/ModelViewer/hooks/useOverlay.ts:33]: Could use filter then map

**Review by Gustav-Eikaas — APPROVED** [review]

File: src/packages/ModelViewer/hooks/useOverlay.ts (hunk 3 above)
Comment (Noggling) [reply src/packages/ModelViewer/hooks/useOverlay.ts:33]: kom med forslag jeg ser ikke hvordan jeg skal få det til uten å få undefined i array.

File: packages/echo3dViewer/src/services/echo3dMultiSelectionHandler.ts (hunk 1 above)
Comment (Noggling) [reply packages/echo3dViewer/src/services/echo3dMultiSelectionHandler.ts:211]: kom med forslag jeg ser ikke hvordan jeg skal få det til uten å få undefined i array.

Comment (Noggling) [reply packages/echo3dViewer/src/services/echo3dMultiSelectionHandler.ts:211]:
``` 
   this.viewerNodeSelection = hierarchyNodeModel.filter((nodeResult) => nodeResult.aabb && nodeResult.tag).map(nodeResult => (
             { position: get3dPositionFromAabbMinMaxValues(nodeResult.aabb!), tagNo: nodeResult.tag!}
                ));
                ```

                dette går men vet ikke jeg

**Pull request merged**